guides/17_Migration_to_3_0.md

```diff
@@ -0,0 +1,120 @@
+# Migration from SDK V2 to SDK V3
+
+## Installation
+
+Installation involves adjusting your `package.json` to depend on version `3.0.0`.
+
+```shell
+npm install --save amazon-chime-sdk-js@3
+```
+
+## Interface changes
+
+__Version 3 of the Amazon Chime SDK for JavaScript makes a number of interface changes.__
+
+__In many cases you should not need to adjust your application code at all. This will be the case if:__
+
+* You do not implement your own `EventController` or construct `DefaultEventController` yourself.
+* You do not pass a `EventReporter` yourself or use the reporter through `MeetingSession`.
+* You do not explicitly call `addObserver` on any instances of `AudioVideoController` using a `AudioVideoObserver` that implements `eventDidReceive`.
+
+If your application does not meet all three criteria, read on.
+
+### Updating `EventController` use cases
+The `EventController` has been decoupled from the `AudioVideoController` which has added addtional required functions to the interface.
+```typescript
+addObserver(observer: EventObserver): void;
+removeObserver(observer: EventObserver): void;
+forEachObserver(observerFunc: (observer: EventObserver) => void): void;
+```
+
+Additionally because of this the constructor of `DefaultEventController` has been updated
+```typescript
+// Before in 2.0:
+constructor(
+    audioVideoController: AudioVideoController
+    eventReporter?: EventReporter
+  )
+
+// After in 3.0:
+constructor(
+    configuration: MeetingSessionConfiguration,
+    logger: Logger,
+    eventReporter?: EventReporter
+  )
+```
+
+### Updating `EventReporter` use cases
+
+The `DefaultMeetingSession` constructor no longer takes in a `EventReporter`.
```

Comment:
In this case, it is more like we take event controller instead of reporter right?